### PR TITLE
fix: validate tensor lengths against cu_seqlens in varlen_attn to prevent NaN gradients (fixes #176793)

### PR DESCRIPTION
**Pull request by Lemniscate-world**
Fixes #176793

## Problem
When padding tokens are added to query/key tensors beyond what \cu_seqlens[-1]\ defines, \arlen_attn()\ completes forward pass without errors but produces NaN gradients in backward. This is a **silent correctness bug** — users get NaN gradients with no error message.

The extra tokens are outside the attention computation but still participate in the autograd graph, causing undefined gradients that propagate through the entire model.

## Fix
Add input validation that raises \ValueError\ with a clear message when:
- \query.size(0) > cu_seq_q[-1]\
- \key.size(0) > cu_seq_k[-1]\

This converts a silent NaN corruption into an explicit, actionable error.

## Reproduction (before fix)
\\\python
cu_seqlens = torch.tensor([0, 144, 432, 944], dtype=torch.int32, device='cuda')
x = torch.randn(946, 1024, device='cuda', requires_grad=True)  # 2 extra tokens
q, k, v = x.chunk(3, dim=-1)
loss = varlen_attn(q, k, v, cu_seqlens, cu_seqlens, 512, 512).sum()
loss.backward()  # NaN in gradients!
\\\

After fix: raises \ValueError\ with clear message pointing to #176793.

## Detection
Detected via [NeuralDBG](https://github.com/LambdaSection/NeuralDBG) — causal diagnostic engine for PyTorch training. The NaN was traced through the gradient health transition system back to \arlen_attn\.

cc @cpuhrsch @drisspg @mikaylagawarecki

### CODE REVIEW

**Comment by Lemniscate-world** [comment]
Friendly ping for review — input validation preventing silent NaN gradients in varlen_attn. Fixes #176793. Thanks!
